### PR TITLE
add support for S3 + ABS filesystems

### PR DESCRIPTION
**Pull request by lwwmanning**
## After this PR
==COMMIT_MSG==
Adds support for Azure & AWS's respective blob stores. Relatedly, upgrade to Java 11 to avoid the many non-performant ciphers on Java 8.
==COMMIT_MSG==

### CODE REVIEW

File: spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/config/HadoopConfiguration.java [hunk 1]
```diff
@@ -36,17 +41,47 @@
...
     Map<String, String> hadoopConf();
 
+    @Value.Default
+    default Map<String, FilesystemConfiguration> filesystems() {
+        return ImmutableMap.of("local", SimpleFilesystemConfiguration.of("file:///tmp/spark-benchmark"));
+    }
+
+    @Value.Default
+    default String defaultFilesystem() {
+        return "local";
```
Comment (rahij) [inline spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/config/HadoopConfiguration.java]: wondering if we need defaults for these in code - feel like it'll be too much magic for users if they don't specify any filesystems (and we ship with default config in the yml anyway)

**Comment by rahij** [comment]
looks good otherwise, will approve once we resolve the comment 👍

File: spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/config/HadoopConfiguration.java (hunk 1 above)
Comment (lwwmanning) [reply spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/config/HadoopConfiguration.java]: done

File: spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/config/HadoopConfiguration.java [hunk 2]
```diff
@@ -41,16 +41,13 @@
 
     Map<String, String> hadoopConf();
 
+    String defaultFilesystem();
+
     @Value.Default
     default Map<String, FilesystemConfiguration> filesystems() {
         return ImmutableMap.of("local", SimpleFilesystemConfiguration.of("file:///tmp/spark-benchmark"));
```
Comment (rahij) [inline spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/config/HadoopConfiguration.java]: this as well?

**Pull request merged**